crates/ra_hir_ty/src/expr.rs

```diff
@@ -235,7 +235,7 @@ pub fn record_pattern_missing_fields(
     pat: &Pat,
 ) -> Option<(VariantId, Vec<LocalStructFieldId>)> {
     let fields = match pat {
-        Pat::Record { path: _, args } => args,
+        Pat::Record { path: _, args, ellipsis: _ } => args,
```

Comment:
I *think* this might be the most convenient change to put under tests. If there's ellibsise, it should return exhaustive=false, like the record literal case about, and it can than be similarly checked in [completion](https://github.com/rust-analyzer/rust-analyzer/blob/080c983498afcac3eb54028af5c9f8bfe7f2c826/crates/ra_ide/src/completion/complete_record.rs#L373-L405).

Long-term, we absolutely need some king of hir diagnostics tests here, but now the completion smoke test seems fitting. cc https://github.com/rust-analyzer/rust-analyzer/pull/3882

Reply:
So I added a bool representing the exhaustiveness to `record_pattern_missing_fields`, similar to  `record_literal_misssing fields`. But `record_pattern_missing_fields` seems to only be called from `crates/ra_hir/src/source_analyzer.rs`, which also calls `record_literal_missing_fields` but ignores the exhaustiveness bool. 

It seems this bool would be useful in creating a `MissingPatField` or similar diagnostic, modeled on the `MissingFields` diagnostic. But I'm not sure how this should be tested without implementing that diagnostic.

Can you provide some more detail on the suggestion of testing this via `completion`?

Reply:
Ah, you are of course right, the presence or absence of `..` doesn't affect completion at all; We just suggest fields not present, and `..` does not change that.  

Yeah, I think the only way to test this now is via a (non-existent) diagnostics.